astr303a.py

```diff
@@ -0,0 +1,571 @@
+    comav = velocity(comameanz)
+
+    #Compute recessional velocities of individual galaxies
+    for i in range(0,len(dataarray)):
+        recev = velocity(dataarray[i][6])
+        pecv = comav-recev
+        dataarray[i].append(pecv)
+
+    #create list of all peculiar velocities
+    pecvslist = []
+    for i in range(0,len(dataarray)):
+        pecvslist.append(dataarray[i][12])
+    
+    #Gather data in bins of width 200km/s. Retrive counts from each bin. 
+    histogram = py.hist(pecvslist,bins=np.arange(-3000, 3000, 200))
+    counts = histogram[0]
+    binvalues = histogram[1]
+    binset = []
+
+    #Create list of peculiar velocities at bin centers
+    for i in range(0,len(binvalues)-1):
+        binset.append(binvalues[i]+100)
+
+    #Get initial parameter guesses (mean,stddev). Get gaussian fit.
+    def gauss(x, a, x0, sigma):
+        return a*np.exp(-(x-x0)**2/(2*sigma**2))
+    stddev = np.std(pecvslist)
+    mean = np.mean(pecvslist)
+    coeffs, pcov = optimize.curve_fit(gauss, binset, counts, p0 = [50, mean, stddev])
+    
+    print('Gaussian fit: '+str(coeffs))
+    print('Velocity dispersion: '+str(coeffs[2])+'km/s')
+    
+    #Plot histogram and gaussian curve.
+    plt.title('Figure 7: Coma Cluster Member Galaxy Peculiar Velocities')
+    plt.xlabel('Peculiar Velocity '+r'$v_{pec}\,(km/s)$', fontsize = 12)
+    plt.ylabel('Number of galaxies', fontsize = 12)
+    plt.xlim(-3000,3000)
+    plt.ylim(0,50)
+    plt.hist(pecvslist,bins=np.arange(-3000, 3000, 200),color='lightgrey',label='Member')
+    plt.plot(binset, gauss(binset, *coeffs),color='black',linestyle='-')
+    pp.savefig()
+    plt.clf()
+    return coeffs,dataarray
+
+#Compute M1/2,L1/2, M/L
+def part161718(cdf,coeffs,reff,rhalf,totallum):
+    mhalf = 3*((coeffs[2]/1.023)**2)*(rhalf*10**6)/(4.5*10**-3)
+    print('Virial mass M1/2: '+str(mhalf)+' M(solar)')
+    lhalf = totallum*(cdf[0]*rhalf**cdf[1] + cdf[2])
+    print('L1/2: '+str(lhalf)+' L(solar)')
+    mlratio = mhalf/lhalf
+    print('M/L: '+str(mlratio))
+    return
+
+######MAIN PROGRAM EXECUTION######
+print('-----------------------------------------------------------------')
+print('                    COMA CLUSTER ANALYSIS')
+print('-----------------------------------------------------------------') 
+dataarray = readData(filename)
+print('')
+part2(dataarray)
+print('')
+comaRA,comaDEC = part3(dataarray)
+print('')
+gausscoeffs = part4(dataarray)
+print('')
+dcoma,zcoma = part5(gausscoeffs,dataarray)
+print('')
+dataarray,dataarray2,dcoma,comaRA,comaDEC,coeffs,comameanz = part678(comaRA,comaDEC,dcoma,dataarray,gausscoeffs,zcoma)
+print('')
+part9(dataarray,dataarray2,comaRA,comaDEC,dcoma,comameanz)
+print('')
+dataarray = part1011(dataarray,dcoma,comaRA,comaDEC)
+print('')
+cdf,reff,rhalf,totallum = part1213(dataarray)
+print('')
+coeffs,dataarray = part1415(dataarray,dcoma,comameanz)
+print('')
+part161718(cdf,coeffs,reff,rhalf,totallum)
+
+pp.close()
+sys.exit()
```

Comment:
Calling `sys.exit()` does nothing here. You have reached the end of your file and that's exactly what happens when the Python interpreter runs out of code to execute.